packages/react/src/components/FileUploader/FileUploader-test.js

```diff
@@ -6,27 +6,86 @@
  */
 
 import React from 'react';
-import { CloseFilled16 } from '@carbon/icons-react';
-import FileUploader, { FileUploaderButton, Filename } from '../FileUploader';
-import FileUploaderSkeleton from '../FileUploader/FileUploader.Skeleton';
-import { mount, shallow } from 'enzyme';
 import { settings } from 'carbon-components';
+import { Close16, CheckmarkFilled16 } from '@carbon/icons-react';
+import { mount, shallow } from 'enzyme';
+import FileUploader, { FileUploaderButton, Filename } from './FileUploader';
+import FileUploaderDropContainer from './FileUploaderDropContainer';
+import FileUploaderItem from './FileUploaderItem';
+import FileUploaderSkeleton from '../FileUploader/FileUploader.Skeleton';
+import InlineLoading from '../InlineLoading';
 
 const { prefix } = settings;
 
 describe('Filename', () => {
-  const mountWrapper = mount(<Filename name={'trees.jpg'} />);
+  describe('Renders as expected', () => {
+    const props = {
+      classNames: [
+        `${prefix}--loading`,
+        `${prefix}--file-close`,
+        `${prefix}--file-complete`,
+      ],
+      icons: [InlineLoading, Close16, CheckmarkFilled16],
+      statuses: ['uploading', 'edit', 'complete'],
+    };
+    props.statuses.forEach((status, i) => {
+      const shallowWrapper = shallow(
+        <Filename iconDescription="Upload complete" status={status} />
+      );
+
+      it('renders upload status icon as expected', () => {
+        expect(shallowWrapper.length).toBe(1);
+        expect(shallowWrapper.find(props.icons[i]).length).toBe(1);
+      });
+    });
+  });
+
+  describe('Check that functions passed in as props are called', () => {
+    const onClick = jest.fn();
+    const onKeyDown = jest.fn();
+    const wrapper = mount(
```

Comment:
We probably shouldn't be running `mount` directly inside of a block, similarly anything that is mocked (like a `jest.fn()`) should be wrapped in a `beforeEach`. Something like:

```js
let mockProps;

beforeEach(() => {
  mockProps = {
    onClick: jest.fn(),
    onKeyDown: jest.fn(),
    status: 'complete',
  };
});

it('should call onClick', () => {
  const wrapper = mount(<Filename {...mockProps} />);
  wrapper.simulate('click');
  expect(onClick).toHaveBeenCalledTimes(1);
});
```

This is because we need to verify that we're correctly setting up / tearing down these mocks in between test invocations otherwise we can get into unexpected states, or are making assertions assuming some state when in reality the tests execute in a way that creates a race condition.